include/circt/Dialect/RTL/RTLStructure.td

```diff
@@ -165,6 +165,110 @@ def RTLModuleExternOp : RTLOp<"module.extern",
   let parser = "return ::parse$cppClass(parser, result);";
 }
 
+def RTLGeneratorSchemaOp : RTLOp<"generator.schema", 
+      [Symbol, HasParent<"mlir::ModuleOp">]> {
+  let summary = "RTL Generator Schema declaration";
+  let description = [{
+    the "rtl.generator.schema" operation declares a kind of generated module by
+    declaring the schema of meta-data required.  
+    A generated module instance of a schema is independent of the external
+    method of producing it.  It is assumed that for well known schema instances,
+    multiple external tools might exist which can process it.  The schema may
+    inherit from another schema.  Generator nodes list attributes required by
+    rtl.module.generated instances and are inherited.
```

Comment:
I'm not totally sure that supporting inheritance here is needed. Or: what is the motivation for supporting inheritance of schemas? (I don't think IR terseness is sufficient justification.)

Reply:
yea, easy enough to add later.